pymeasure/instruments/keithley/keithley2182.py

```diff
@@ -0,0 +1,399 @@
+                                              strict_range)
+
+from .buffer import KeithleyBuffer
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class Keithley2182Channel(Channel):
+    """Implementation of a Keithley 2182 channel.
+
+    Channel 1 is the fundamental measurement channel, while channel 2 provides
+    sense measurements. Channel 2 inputs are referenced to Channel 1 LO.
+    Possible configurations are
+
+    Voltage     (Channel 1)
+    Temperature (Channel 1)
+    Voltage     (Channel 1) and Voltage     (Channel 2)
+    Voltage     (Channel 1) and Temperature (Channel 2)
+    """
+
+    def __init__(self, instrument, id_):
+        """Set max voltage depending on channel."""
+        if id_ == 1:
+            self.voltage_range_values = [0, 120]
+            self.voltage_offset_values = [-120, 120]
+        else:
+            self.voltage_range_values = [0, 12]
+            self.voltage_offset_values = [-12, 12]
+        super().__init__(instrument, id_)
+
+    voltage_range = Channel.control(
+        ":SENS:VOLT:CHAN{ch}:RANG?", ":SENS:VOLT:CHAN{ch}:RANG %g",
+        """Control the positive full-scale measurement voltage range in Volts.
+        Valid ranges are from 0 to 120 V for Ch. 1, and 0 to 12 V for Ch. 2.
+        Auto-range is automatically disabled when this property is set.""",
+        validator=truncated_range,
+        values=[0, 120],
+        dynamic=True,
+    )
+    voltage_range_auto = Channel.control(
+        ":SENS:VOLT:CHAN{ch}:RANG:AUTO?", ":SENS:VOLT:CHAN{ch}:RANG:AUTO %d",
+        """Control whether auto voltage ranging is enabled.
+        Valid values are True and False.""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+    voltage_offset = Channel.control(
+        ":SENS:VOLT:CHAN{ch}:REF?", ":SENS:VOLT:CHAN{ch}:REF %g",
+        """Control the relative offset for measuring voltage.
+        Displayed value = actual value - offset value.
+        Valid ranges are -120 V to +120 V for Ch. 1, and -12 V to +12 V for Ch. 2.""",
+        validator=strict_range,
+        values=[-120, 120],
+        dynamic=True,
+    )
+    temperature_offset = Channel.control(
+        ":SENS:TEMP:CHAN{ch}:REF?", ":SENS:TEMP:CHAN{ch}:REF %g",
+        """Control the relative offset for measuring temperature.
+        Displayed value = actual value - offset value.
+        Valid values are -273 C to 1800 C.""",
+        validator=strict_range,
+        values=[-273, 1800],
+    )
+    voltage_offset_state = Channel.control(
+        ":SENS:VOLT:CHAN{ch}:REF:STAT?", ":SENS:VOLT:CHAN{ch}:REF:STAT %s",
+        """Control whether voltage is measured as a relative or absolute value.
+        Enabled by default for Ch. 2 voltage, which is measured relative to Ch. 1
+        voltage.""",
+        validator=strict_discrete_set,
+        values={'OFF': 0, 'ON': 1},
+        map_values=True
+    )
```

Comment:
Why don't you make it boolean (also changing the name)?
```suggestion
    voltage_offset_enabled = Channel.control(
        ":SENS:VOLT:CHAN{ch}:REF:STAT?", ":SENS:VOLT:CHAN{ch}:REF:STAT %s",
        """Control whether voltage is measured as a relative or absolute value.
        Enabled by default for Ch. 2 voltage, which is measured relative to Ch. 1
        voltage (bool).""",
        validator=strict_discrete_set,
        values={False: 0, True: 1},
        map_values=True
    )
```

similarly below.